processor/k8sattributesprocessor/internal/metadata/generated_resource.go

```diff
@@ -0,0 +1,174 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package metadata
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+// ResourceBuilder is a helper struct to build resources predefined in metadata.yaml.
+type ResourceBuilder struct {
+	config ResourceAttributesConfig
+	res    pcommon.Resource
+}
+
+func NewResourceBuilder(rac ResourceAttributesConfig) *ResourceBuilder {
+	return &ResourceBuilder{
+		config: rac,
+		res:    pcommon.NewResource(),
+	}
+}
+
+// SetContainerID sets provided value as "container.id" attribute.
+func (rb *ResourceBuilder) SetContainerID(val string) {
+	if rb.config.ContainerID.Enabled {
+		rb.res.Attributes().PutStr("container.id", val)
+	}
+}
```

Comment:
I would follow the same approach we have for the metrics from the beginning. And if we want to add this one later, we can do it for both. Resource attributes are being set only once, so the result of this optimization is negligible, but user's ergonomist can be worse for most of the cases

Reply:
> Resource attributes are being set only once, so the result of this optimization is negligible

This could be important for reducing the start time of the Collector (which some people care about), since gathering the resource attributes is done during startup. But I guess it's fine to leave it for later since we can always duplicate the methods.